### PR TITLE
Fixed error while trying to serialize empty messages.

### PR DESCRIPTION
**Pull request by hveiras**
connects to #140

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@hveiras__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: src/Implementation/WarmStorage/WarmStorage/WarmStorageProcessor.cs [hunk 1]
```diff
@@ -50,7 +50,10 @@ public Task CloseAsync(PartitionContext context, CloseReason reason)
         public async Task ProcessEventsAsync(PartitionContext context, IEnumerable<EventData> events)
         {
             // Workaround for event hub sending null on timeout
-            events = events ?? Enumerable.Empty<EventData>();
+            if(events == null)
+            {
+                return;
+            }
```
Comment (bennage) [inline src/Implementation/WarmStorage/WarmStorage/WarmStorageProcessor.cs]: What's the motivation for this change? To exit execution a bit earlier?

File: src/Implementation/WarmStorage/WarmStorage/WarmStorageCoordinator.cs [hunk 2]
```diff
@@ -56,8 +56,7 @@ private WarmStorageCoordinator(EventProcessorHost host)
             {
                 MaxBatchSize = configuration.MaxBatchSize,
                 PrefetchCount = configuration.PreFetchCount,
-                ReceiveTimeOut = configuration.ReceiveTimeout,
-                InvokeProcessorAfterReceiveTimeout = true
+                ReceiveTimeOut = configuration.ReceiveTimeout
```
Comment (bennage) [inline src/Implementation/WarmStorage/WarmStorage/WarmStorageCoordinator.cs]: In previous releases, `ReceiveTimeOut` was ignored unless `InvokeProcessorAfterReceiveTimeout` was set to true. Has this changed in the current release?

Comment (hveiras) [inline src/Implementation/WarmStorage/WarmStorage/WarmStorageCoordinator.cs]: I tried removing InvokeProcessorAfterReceiveTimeout (the default is false) and EH still called me  with null events on timeout. That is why I added the check anyways.

Comment (bennage) [inline src/Implementation/WarmStorage/WarmStorage/WarmStorageCoordinator.cs]: Hmm... interesting. I wonder is setting `ReceiveTimeout` is what triggers it now.
I'm still curious to know why your replaced 
``` C#
events = events ?? Enumerable.Empty<EventData>();
```
It was a guard against the null with the hope that one day the bug (I think that passing a null here is a bug) would be replaced with an empty enumerable.

Comment (hveiras) [inline src/Implementation/WarmStorage/WarmStorage/WarmStorageCoordinator.cs]: You are right I haven't thought about that. Changing it...
